serving/src/main/java/ai/djl/serving/http/AdapterManagementRequestHandler.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.http;
+
+import ai.djl.ModelException;
+import ai.djl.modality.Input;
+import ai.djl.modality.Output;
+import ai.djl.serving.http.list.ListAdaptersResponse;
+import ai.djl.serving.http.list.ListPagination;
+import ai.djl.serving.models.ModelManager;
+import ai.djl.serving.util.NettyUtils;
+import ai.djl.serving.wlm.Adapter;
+import ai.djl.serving.wlm.ModelInfo;
+import ai.djl.serving.wlm.WorkerPool;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/** A class handling inbound HTTP requests to the management API for adapters. */
+public class AdapterManagementRequestHandler extends HttpRequestHandler {
+    private static final Pattern ADAPTERS_PATTERN =
+            Pattern.compile("^/models/.+/adapters([/?][^/]*)?");
```

Comment:
Yeah, I will put documents in the next PR because there is a good amount required:

- Document HTTP adapters management API
- Document adapter usage
- Document adapter creation
- Document adapter options and configurations